packages/protvista-datatable/src/protvista-datatable.ts

```diff
@@ -224,39 +206,58 @@ class ProtvistaDatatable<T extends DataTableDatum> extends LitElement {
     );
   }
 
-  getStyleClass(id: string, start: number, end: number): string {
-    let className = "";
-    if (this.selectedid && this.selectedid === id) {
-      className = `${className} active`;
-    }
-    if (
-      this.displayStart &&
-      this.displayEnd &&
-      ProtvistaDatatable.isOutside(
-        this.displayStart,
-        this.displayEnd,
-        Number(start),
-        Number(end)
-      )
-    ) {
-      className = `${className} hidden`;
-    }
-    if (
-      this.highlight &&
-      ProtvistaDatatable.isWithinRange(
-        this.highlight[0],
-        this.highlight[1],
-        Number(start),
-        Number(end)
-      )
-    ) {
-      className = `${className} overlapped`;
-    }
-    return className;
-  }
+  updateRowStyling(): void {
+    let oddOrEvenCount = 0;
+    this.rows?.forEach((row) => {
+      const { start, end } = row.dataset;
+      row.classList.add(oddOrEvenCount % 2 === 0 ? "odd" : "even");
```

Comment:
Is this because you add rows in the middle for the extra content that you cannot do that only with CSS? Not sure if you'll filter rows at some point, but I guess you'll have to remove any existing class and re-set it whenever you do, o whenever the source table is modified

Reply:
Also, the end result will be the same just inverted, but the code should probably be `oddOrEvenCount % 2 === 0 ? "even" : "odd"` in order to match what the logic says

Reply:
> Is this because you add rows in the middle for the extra content that you cannot do that only with CSS?  
Yes. Can't combine `nth-child` with `not` because of CSS's fully declarative nature.
And you're correct about having to update it after filtering.

> in order to match what the logic says
Will update the logic to be the right way round